weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CompareEvaluationsPage/sections/ExampleCompareSection/ExampleCompareSection.tsx

```diff
@@ -250,7 +262,8 @@ export const ExampleCompareSection: React.FC<{
     k => k !== DERIVED_SCORER_REF_PLACEHOLDER
   );
 
-  const inputRef = parseRef(target.inputRef) as WeaveObjectRef;
+  // Handle input reference null case
+  const inputRef = safeParseRef(target.inputRef);
```

Comment:
this pattern is a smell. Instead of having `inputRef` acting like "maybeInputRef", I would modify the data structure itself such that if `inputRef` is present, it is pre-checked and indeed an input ref. Should this be an optional field? Or maybe a discriminated union?

Reply:
modify the data structure of `target`?

Reply:
yeah - target.inputRef can be null now. But you can be sure that if it is not null, it is indeed a ref string